Sources/Basic/Path.swift

```diff
@@ -118,12 +124,31 @@ public struct AbsolutePath {
         return AbsolutePath(self, subpath)
     }
     
-    /// Returns the absolute path with the contents of the string (interpreted
-    /// as a relative path, not a single path component) appended.
-    public func appending(_ str: String) -> AbsolutePath {
-        return AbsolutePath(self, str)
+    /// Returns the absolute path with an additional literal component appended.
+    ///
+    /// This method should only be used in cases where the input is guaranteed
+    /// to be a valid path component (i.e., it cannot be empty, contain a path
+    /// separator, or be a pseudo-path like '.' or '..').
+    public func appending(component name: String) -> AbsolutePath {
+        assert(name != "" && name != "." && name != ".." && !name.contains("/"))
```

Comment:
If this method asserts rather than throws an error, it seems to me that we also need to add a method to check if a particular string is a valid path component.  Otherwise, a caller can't effectively check input strings, and the result isn't an error that can be caught and displayed but instead just causes a crash.  That seems unfortunate.